static/js/containers/CreatePostPage.js

```diff
@@ -51,6 +61,32 @@ class CreatePostPage extends React.Component<*, void> {
       })
     )
     ensureTwitterEmbedJS()
+    this.updateTabSelection()
+  }
+
+  componentDidUpdate() {
+    this.updateTabSelection()
+  }
+
+  updateTabSelection = () => {
+    const { channel, dispatch, postForm } = this.props
+
+    if (
+      postForm &&
+      channel &&
+      (postForm.value.postType === null ||
+        !isLinkTypeChecked(channel.link_type, postForm.value.postType))
+    ) {
+      const postType = isTextTabSelected(null, channel)
+        ? LINK_TYPE_TEXT
+        : LINK_TYPE_LINK
```

Comment:
reading through this is a bit difficult to follow, since `isTextTabSelected` is called in a bunch of places for different functions. it seems that what we're doing here is check if the form postType value is either null or a value which is not allowed for this channel, if it's not allowed for the channel we want to set it to the right one.

I think it may make sense to define another function, other than `isLinkTypeChecked`, which is called `isLinkTypeAllowed` - it may check a very similar, or even the same, thing, but it can improve readability by making it more clear what we're doing. It might also be more readable to just check

```js
(postForm.value.postType === null ||
  (channel.link_type !== LINK_TYPE_ANY && (channel.link_type !== postForm.value.postType)))
```
just mentioning this because when reading this it took me a while to read through the different functions, parse out what they were doing with null arguments, etc.

also, in this case, I *think*, if I'm reading things right, we can set the `postType` variable here to be

```js
const postType = channel.link_type === LINK_TYPE_ANY
  ? LINK_TYPE_TEXT
  : channel.link_type
```

if the channel link type is `LINK_TYPE_ANY`, then we're in this block because `postForm.value.postType` is `null`, so we can just pick `LINK_TYPE_TEXT` as a default value. On the other hand, if the `channel.link_type` is not `LINK_TYPE_ANY`, then the value of `channel.link_type` is going to be the only valid choice anyhow.